### DISCUSSION TITLE
Can't fetch auth user?

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#6640] [Questions] [answered]
![Ekran Resmi 2022-05-01 15 02 18](https://user-images.githubusercontent.com/42077752/166144561-79143efc-4fb1-4145-8655-9b6fd81a3240.png)
![Ekran Resmi 2022-05-01 15 07 08](https://user-images.githubusercontent.com/42077752/166144700-5e390207-0c95-4568-b014-cbde798c4128.png)

When i fetch auth user fetch comes null, i checked before with .single() than i got error
how to solve it?

**Comment by GaryAustin1** [comment, marked as answer]
If this call is done when you start the app it can take a bit for auth to get setup.  You can add onAuthStateChange to see if signin event has occurred before your call.  https://supabase.com/docs/reference/javascript/auth-onauthstatechange

If you have RLS on and don’t meet policy then data also comes back as empty array.

**Reply by ghost** [reply]
Actually there is like this

`  const [session, setSession] = useState(null)

  useEffect(() => {
    setSession(supabase.auth.session())

    supabase.auth.onAuthStateChange((_event, session) => {
      setSession(session)
    })
  }, [])`

**Reply by GaryAustin1** [reply]
OK, if you have a console.log in that handler does signin occur before your database call?
Do you have RLS on and if so what is the policy.

**Reply by ghost** [reply]
![Ekran Resmi 2022-05-01 19 02 11](https://user-images.githubusercontent.com/42077752/166153796-d5c529bd-5adc-4678-87c1-e3a6d265d8d3.png)
![Ekran Resmi 2022-05-01 19 05 37](https://user-images.githubusercontent.com/42077752/166153886-40912d05-38d2-4d36-bedd-a43f03de05a0.png)

Its sign in and also there are policies

**Reply by GaryAustin1** [reply]
Hmmm.  Your policy should not be an issue, even if user was not logged in.  
If you hard code a uid in .eq('id','useridstring') does that work?  Does the uid exist in the table?

Edit, I'm not sure you have shown you are actually signed in and have a user when you use auth.user() in your comments so far.  Local storage can be set, but until the client gets set up and issues the signin event there is no user in auth.user() yet.

**Comment by ghost** [comment]
![Ekran Resmi 2022-05-01 19 35 53](https://user-images.githubusercontent.com/42077752/166155330-6d8cda4e-564e-419f-bcdd-ac22e72b62ac.png)
![Ekran Resmi 2022-05-01 19 44 30](https://user-images.githubusercontent.com/42077752/166155332-6dcc07a5-0b44-491e-90fc-3b399a94ca25.png)

With string uid it worked i got user data but user.id still not working.

How to set Signin event?

**Reply by GaryAustin1** [reply]
Seems like you are trying to use the user object before supabase.js is finished setting up and issuing a signin event.  Even if in local storage and logged in, it takes some time.

https://supabase.com/docs/reference/javascript/auth-onauthstatechange#listen-to-sign-in

**Reply by ghost** [reply]
Is there any place that how to implement authstate Signin, because i dont know where to put that?

**Comment by GaryAustin1** [comment]
I don't use React anymore, but in the React example https://supabase.com/docs/guides/with-react it watches for the session changing like you do in onAuthStateChange and then in the Account page it does this:
```  
useEffect(() => {
    getProfile()
  }, [session])
```
So it does not fetch the profile until it gets a session change (which would occur on signin or signout)

If you plan on doing server side stuff in your application they you might want to consider using auth-helpers https://github.com/supabase-community/supabase-auth-helpers

**Reply by ghost** [reply]
Thank you very much for your help, what do you use instead of react?

**Reply by GaryAustin1** [reply]
I use SvelteKit now.  For whatever reason, I just find it easier to deal with states, stores and components.
